File: ui/lib/types/sso.ts
import type { EnvVar } from "./schemas";

export interface GoogleSSOConfig {
	client_id: EnvVar;
	client_secret: EnvVar;
	allowed_domains?: string[];
}

export interface SAMLConfig {
	entity_id: string;
	metadata_url?: string;
	idp_sso_url?: string;
	idp_certificate?: string;
	idp_entity_id?: string;
	sign_requests: boolean;
	force_authn: boolean;
	name_id_format?: string;
}

export interface User {
	id: string;
	email: string;
	name: string;
	role: string; // "admin" | "viewer"
	auth_provider: string; // "password" | "google_sso" | "saml"
	external_id: string;
	avatar_url: string;
	is_active: boolean;
	last_login_at: string | null;
	created_at: string;
	updated_at: string;
}

export interface AuthMethodsResponse {
	is_auth_enabled: boolean;
	has_valid_token: boolean;
	enabled_methods: string[];
	google_sso?: { login_url: string };
	saml?: { login_url: string };
}

File: core/schemas/sso.go
package schemas

// GoogleSSOConfig holds configuration for Google OAuth SSO login.
type GoogleSSOConfig struct {
	ClientID     *EnvVar `json:"client_id"`
	ClientSecret *EnvVar `json:"client_secret"`
	RedirectURI  string  `json:"redirect_uri,omitempty"`
}

// SAMLConfig holds configuration for SAML-based SSO login.
type SAMLConfig struct {
	IDPURL      string `json:"idp_url"`
	Certificate string `json:"certificate,omitempty"`
	EntityID    string `json:"entity_id,omitempty"`
}
